apps/systatum/i18n/LocalizeProvider.tsx

```diff
@@ -0,0 +1,32 @@
+"use client";
+
+import { useLocaleStore } from "@/hooks/contents/useLocaleStore";
+
+export default function LocaleProvider({ locale }: { locale: string }) {
+  initializeLocale(locale);
+
+  return null;
+}
+
+export const initializeLocale = (localeFromServer?: string) => {
+  if (!localeFromServer) return;
+  const code = localeFromServer;
+
+  useLocaleStore.setState((prev) => ({
+    ...prev,
+    locale: {
+      code,
+      label:
```

Comment:
can we extract it out and use `switch` :::::::: << risk of too many of that someday I am afraid :D